File: Assets/Scripts/Countdown.js
#pragma strict

private var timerText : UI.Text;
private var canvasObj : GameObject;
private var child : Transform;

function Start () {
	canvasObj = GameObject.Find("Canvas");
	child = canvasObj.transform.Find("lblTime");
	timerText = child.GetComponent(UI.Text);
	InvokeRepeating("TimerTick", 0, 1f);
}

function TimerTick() {
	if(GameVariables.currentTime > 0) {
		GameVariables.currentTime--;
		timerText.text = "Time : " + GameVariables.currentTime.ToString();
	} else {
		// game over and restart
		SceneManagement.SceneManager.LoadScene("Scene1Exterieur");
	}
}

function CancelTimer() {
	CancelInvoke("TimerTick");
}

File: Assets/Scripts/Countdown.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;
using UnityEngine.SceneManagement;

public class Countdown : MonoBehaviour {
	private Text timerText;
	private GameObject canvasObj;
	private Transform child;

	void Start () {
		canvasObj = GameObject.Find("Canvas");
		child = canvasObj.transform.Find("lblTime");
		timerText = child.GetComponent<Text>();
		InvokeRepeating("TimerTick", 0, 1f);
	}

	void TimerTick() {
		if(GameVariables.currentTime > 0) {
			GameVariables.currentTime--;
			timerText.text = "Time : " + GameVariables.currentTime.ToString();
		} else {
			// game over and restart
			if(SceneManager.GetActiveScene().name.Equals("Scene2Interieur")) {
				Destroy(GameObject.FindGameObjectWithTag("GameMusic"));
			}
			GameVariables.currentTime = GameVariables.allowedTime;
			SceneManager.LoadScene("Scene1Exterieur");
		}
	}

	void CancelTimer() {
		CancelInvoke("TimerTick");
		timerText.text = "Time : " + GameVariables.currentTime.ToString();
	}
}